client/src/variants/Remarkablerookies.js

```diff
@@ -0,0 +1,205 @@
+import { ChessRules } from "@/base_rules";
+
+export class RemarkablerookiesRules extends ChessRules {
+
+  static get PawnSpecs() {
+    return Object.assign(
+      {},
+      ChessRules.PawnSpecs,
+      {
+        promotions:
+          ChessRules.PawnSpecs.promotions.concat(
+          [V.SHORT_ROOK, V.WOODY_ROOK, V.HALF_DUCK, V.CHANCELLOR])
+      }
+    );
+  }
+
+  getPpath(b) {
+    if ([V.SHORT_ROOK, V.WOODY_ROOK, V.HALF_DUCK, V.CHANCELLOR].includes(b[1]))
+      return "Remarkablerookies/" + b;
+    return b;
+  }
+
+  static GenRandInitFen(randomness) {
+    if (randomness == 0)
+      return "dhaskahd/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w 0 ahah -";
+
+    // Mapping white --> black (at least at start):
+    const piecesMap = {
+      'r': 'd',
+      'n': 'h',
+      'b': 'a',
+      'q': 's',
+      'k': 'k'
+    };
+
+    const baseFen = ChessRules.GenRandInitFen(randomness);
+    return (
+      baseFen.substr(0, 8).split('').map(p => piecesMap[p]).join('') +
+      baseFen.substr(8)
+    );
+  }
+
+  static get SHORT_ROOK() {
+    return 'd';
+  }
+  static get WOODY_ROOK() {
+    return 'h';
+  }
+  static get HALF_DUCK() {
+    return 'a';
+  }
+  static get CHANCELLOR() {
+    return 's';
```

Comment:
These piece abbreviations are simply because I used Colorbound.js  as a template -- no particular thought went into them :)

If you would want to move forward with this variant, it probably would be worth me changing the abbreviations to something that makes sense.